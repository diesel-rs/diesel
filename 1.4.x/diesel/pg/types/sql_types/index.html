<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="PostgreSQL specific SQL types"><meta name="keywords" content="rust, rustlang, rust-lang, sql_types"><title>diesel::pg::types::sql_types - Rust</title><link rel="stylesheet" type="text/css" href="../../../../normalize.css"><link rel="stylesheet" type="text/css" href="../../../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../../../ayu.css" disabled ><script id="default-settings" ></script><script src="../../../../storage.js"></script><script src="../../../../crates.js"></script><noscript><link rel="stylesheet" href="../../../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../../../favicon.svg"><link rel="alternate icon" type="image/png" href="../../../../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../../../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../../../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu" role="button">&#9776;</div><a href='../../../../diesel/index.html'><div class='logo-container rust-logo'><img src='../../../../rust-logo.png' alt='logo'></div></a><h2 class="location">Module sql_types</h2><div class="sidebar-elems"><div class="block items"><ul><li><a href="#structs">Structs</a></li><li><a href="#types">Type Definitions</a></li></ul></div><div id="sidebar-vars" data-name="sql_types" data-ty="mod" data-relpath="./"></div><script defer src="./sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!" aria-haspopup="menu" title="themes"><img width="18" height="18" alt="Pick another theme!" src="../../../../brush.svg"></button><div id="theme-choices" role="menu"></div></div><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input"name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><button type="button" id="help-button" title="help">?</button><a id="settings-menu" href="../../../../settings.html" title="settings"><img width="18" height="18" alt="Change settings" src="../../../../wheel.svg"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="in-band">Module <a href="../../../index.html">diesel</a>::<wbr><a href="../../index.html">pg</a>::<wbr><a href="../index.html">types</a>::<wbr><a class="mod" href="#">sql_types</a><button id="copy-path" onclick="copy_path(this)" title="copy path"><img src="../../../../clipboard.svg" width="19" height="18" alt="Copy item import" title="Copy item import to clipboard"></button></span><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../../../src/diesel/pg/types/mod.rs.html#27-493" title="goto source code">[src]</a></span></h1><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>PostgreSQL specific SQL types</p>
<p>Note: All types in this module can be accessed through <code>diesel::sql_types</code></p>
</div></details><h2 id="structs" class="section-header"><a href="#structs">Structs</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="struct" href="struct.Array.html" title="diesel::pg::types::sql_types::Array struct">Array</a></div><div class="item-right docblock-short"><p>The <code>Array</code> SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Cidr.html" title="diesel::pg::types::sql_types::Cidr struct">Cidr</a></div><div class="item-right docblock-short"><p>The <a href="https://www.postgresql.org/docs/9.6/static/datatype-net-types.html"><code>CIDR</code></a> SQL type. This type can only be used with <code>feature = &quot;network-address&quot;</code></p>
</div><div class="item-left module-item"><a class="struct" href="struct.Inet.html" title="diesel::pg::types::sql_types::Inet struct">Inet</a></div><div class="item-right docblock-short"><p>The <a href="https://www.postgresql.org/docs/9.6/static/datatype-net-types.html"><code>INET</code></a> SQL type. This type can only be used with <code>feature = &quot;network-address&quot;</code></p>
</div><div class="item-left module-item"><a class="struct" href="struct.Json.html" title="diesel::pg::types::sql_types::Json struct">Json</a></div><div class="item-right docblock-short"><p>The JSON SQL type.  This type can only be used with <code>feature = &quot;serde_json&quot;</code></p>
</div><div class="item-left module-item"><a class="struct" href="struct.Jsonb.html" title="diesel::pg::types::sql_types::Jsonb struct">Jsonb</a></div><div class="item-right docblock-short"><p>The <code>jsonb</code> SQL type.  This type can only be used with <code>feature = &quot;serde_json&quot;</code></p>
</div><div class="item-left module-item"><a class="struct" href="struct.MacAddr.html" title="diesel::pg::types::sql_types::MacAddr struct">MacAddr</a></div><div class="item-right docblock-short"><p>The <a href="https://www.postgresql.org/docs/9.6/static/datatype-net-types.html"><code>MACADDR</code></a> SQL type. This type can only be used with <code>feature = &quot;network-address&quot;</code></p>
</div><div class="item-left module-item"><a class="struct" href="struct.Money.html" title="diesel::pg::types::sql_types::Money struct">Money</a></div><div class="item-right docblock-short"><p>The PostgreSQL <a href="https://www.postgresql.org/docs/9.1/static/datatype-money.html">Money</a> type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Oid.html" title="diesel::pg::types::sql_types::Oid struct">Oid</a></div><div class="item-right docblock-short"><p>The <code>OID</code> SQL type. This is a PostgreSQL specific type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Range.html" title="diesel::pg::types::sql_types::Range struct">Range</a></div><div class="item-right docblock-short"><p>The <code>Range</code> SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Record.html" title="diesel::pg::types::sql_types::Record struct">Record</a></div><div class="item-right docblock-short"><p>The <code>Record</code> (a.k.a. tuple) SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Timestamptz.html" title="diesel::pg::types::sql_types::Timestamptz struct">Timestamptz</a></div><div class="item-right docblock-short"><p>The “timestamp with time zone” SQL type, which PostgreSQL abbreviates
to <code>timestamptz</code>.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Uuid.html" title="diesel::pg::types::sql_types::Uuid struct">Uuid</a></div><div class="item-right docblock-short"><p>The <code>UUID</code> SQL type. This type can only be used with <code>feature = &quot;uuid&quot;</code></p>
</div></div><h2 id="types" class="section-header"><a href="#types">Type Definitions</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="type" href="type.BigSerial.html" title="diesel::pg::types::sql_types::BigSerial type">BigSerial</a></div><div class="item-right docblock-short"><p>Alias for <code>BigInt</code></p>
</div><div class="item-left module-item"><a class="type" href="type.Serial.html" title="diesel::pg::types::sql_types::Serial type">Serial</a></div><div class="item-right docblock-short"><p>Alias for <code>Integer</code></p>
</div><div class="item-left module-item"><a class="type" href="type.SmallSerial.html" title="diesel::pg::types::sql_types::SmallSerial type">SmallSerial</a></div><div class="item-right docblock-short"><p>Alias for <code>SmallInt</code></p>
</div></div></section><section id="search" class="content hidden"></section><div id="rustdoc-vars" data-root-path="../../../../" data-current-crate="diesel" data-search-index-js="../../../../search-index.js" data-search-js="../../../../search.js"></div>
    <script src="../../../../main.js"></script>
</body></html>