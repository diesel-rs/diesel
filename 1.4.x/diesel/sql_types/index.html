<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="Types which represent a SQL data type."><meta name="keywords" content="rust, rustlang, rust-lang, sql_types"><title>diesel::sql_types - Rust</title><link rel="stylesheet" type="text/css" href="../../normalize.css"><link rel="stylesheet" type="text/css" href="../../rustdoc.css" id="mainThemeStyle"><link rel="stylesheet" type="text/css" href="../../light.css"  id="themeStyle"><link rel="stylesheet" type="text/css" href="../../dark.css" disabled ><link rel="stylesheet" type="text/css" href="../../ayu.css" disabled ><script id="default-settings" ></script><script src="../../storage.js"></script><script src="../../crates.js"></script><noscript><link rel="stylesheet" href="../../noscript.css"></noscript><link rel="icon" type="image/svg+xml" href="../../favicon.svg"><link rel="alternate icon" type="image/png" href="../../favicon-16x16.png"><link rel="alternate icon" type="image/png" href="../../favicon-32x32.png"><style type="text/css">#crate-search{background-image:url("../../down-arrow.svg");}</style></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="sidebar"><div class="sidebar-menu" role="button">&#9776;</div><a href='../../diesel/index.html'><div class='logo-container rust-logo'><img src='../../rust-logo.png' alt='logo'></div></a><h2 class="location">Module sql_types</h2><div class="sidebar-elems"><div class="block items"><ul><li><a href="#modules">Modules</a></li><li><a href="#structs">Structs</a></li><li><a href="#traits">Traits</a></li><li><a href="#types">Type Definitions</a></li></ul></div><div id="sidebar-vars" data-name="sql_types" data-ty="mod" data-relpath="./"></div><script defer src="./sidebar-items.js"></script></div></nav><div class="theme-picker"><button id="theme-picker" aria-label="Pick another theme!" aria-haspopup="menu" title="themes"><img width="18" height="18" alt="Pick another theme!" src="../../brush.svg"></button><div id="theme-choices" role="menu"></div></div><nav class="sub"><form class="search-form"><div class="search-container"><div><select id="crate-search"><option value="All crates">All crates</option></select><input class="search-input"name="search" disabled autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"></div><button type="button" id="help-button" title="help">?</button><a id="settings-menu" href="../../settings.html" title="settings"><img width="18" height="18" alt="Change settings" src="../../wheel.svg"></a></div></form></nav><section id="main" class="content"><h1 class="fqn"><span class="in-band">Module <a href="../index.html">diesel</a>::<wbr><a class="mod" href="#">sql_types</a><button id="copy-path" onclick="copy_path(this)" title="copy path"><img src="../../clipboard.svg" width="19" height="18" alt="Copy item import" title="Copy item import to clipboard"></button></span><span class="out-of-band"><span id="render-detail"><a id="toggle-all-docs" href="javascript:void(0)" title="collapse all docs">[<span class="inner">&#x2212;</span>]</a></span><a class="srclink" href="../../src/diesel/sql_types/mod.rs.html#1-499" title="goto source code">[src]</a></span></h1><details class="rustdoc-toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>Types which represent a SQL data type.</p>
<p>The structs in this module are <em>only</em> used as markers to represent a SQL type.
They should never be used in your structs.
If you’d like to know the rust types which can be used for a given SQL type,
see the documentation for that SQL type.
Additional types may be provided by other crates.</p>
<p>To see which SQL type can be used with a given Rust type,
see the “Implementors” section of <a href="../deserialize/trait.FromSql.html"><code>FromSql</code></a>.</p>
<p>Any backend specific types are re-exported through this module</p>
</div></details><h2 id="reexports" class="section-header"><a href="#reexports">Re-exports</a></h2>
<div class="item-table"><div class="item-left import-item"><code>pub use pg::types::<a class="mod" href="../pg/types/sql_types/index.html" title="mod diesel::pg::types::sql_types">sql_types</a>::*;</code></div><div class="item-right docblock-short"></div><div class="item-left import-item"><code>pub use mysql::<a class="mod" href="../mysql/types/index.html" title="mod diesel::mysql::types">types</a>::*;</code></div><div class="item-right docblock-short"></div></div><h2 id="modules" class="section-header"><a href="#modules">Modules</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="mod" href="ops/index.html" title="diesel::sql_types::ops mod">ops</a></div><div class="item-right docblock-short"><p>Represents the output of numeric operators in SQL</p>
</div></div><h2 id="structs" class="section-header"><a href="#structs">Structs</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="struct" href="struct.BigInt.html" title="diesel::sql_types::BigInt struct">BigInt</a></div><div class="item-right docblock-short"><p>The big integer SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Binary.html" title="diesel::sql_types::Binary struct">Binary</a></div><div class="item-right docblock-short"><p>The binary SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Bool.html" title="diesel::sql_types::Bool struct">Bool</a></div><div class="item-right docblock-short"><p>The boolean SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Date.html" title="diesel::sql_types::Date struct">Date</a></div><div class="item-right docblock-short"><p>The date SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Double.html" title="diesel::sql_types::Double struct">Double</a></div><div class="item-right docblock-short"><p>The double precision float SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Float.html" title="diesel::sql_types::Float struct">Float</a></div><div class="item-right docblock-short"><p>The float SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Integer.html" title="diesel::sql_types::Integer struct">Integer</a></div><div class="item-right docblock-short"><p>The integer SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Interval.html" title="diesel::sql_types::Interval struct">Interval</a></div><div class="item-right docblock-short"><p>The interval SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Nullable.html" title="diesel::sql_types::Nullable struct">Nullable</a></div><div class="item-right docblock-short"><p>The nullable SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Numeric.html" title="diesel::sql_types::Numeric struct">Numeric</a></div><div class="item-right docblock-short"><p>The arbitrary precision numeric SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.SmallInt.html" title="diesel::sql_types::SmallInt struct">SmallInt</a></div><div class="item-right docblock-short"><p>The small integer SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Text.html" title="diesel::sql_types::Text struct">Text</a></div><div class="item-right docblock-short"><p>The text SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Time.html" title="diesel::sql_types::Time struct">Time</a></div><div class="item-right docblock-short"><p>The time SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.Timestamp.html" title="diesel::sql_types::Timestamp struct">Timestamp</a></div><div class="item-right docblock-short"><p>The timestamp SQL type.</p>
</div><div class="item-left module-item"><a class="struct" href="struct.TinyInt.html" title="diesel::sql_types::TinyInt struct">TinyInt</a></div><div class="item-right docblock-short"><p>The tiny integer SQL type.</p>
</div></div><h2 id="traits" class="section-header"><a href="#traits">Traits</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="trait" href="trait.Foldable.html" title="diesel::sql_types::Foldable trait">Foldable</a></div><div class="item-right docblock-short"><p>Represents SQL types which can be used with <code>SUM</code> and <code>AVG</code></p>
</div><div class="item-left module-item"><a class="trait" href="trait.HasSqlType.html" title="diesel::sql_types::HasSqlType trait">HasSqlType</a></div><div class="item-right docblock-short"><p>Indicates that a SQL type exists for a backend.</p>
</div><div class="item-left module-item"><a class="trait" href="trait.IntoNullable.html" title="diesel::sql_types::IntoNullable trait">IntoNullable</a></div><div class="item-right docblock-short"><p>Converts a type which may or may not be nullable into its nullable
representation.</p>
</div><div class="item-left module-item"><a class="trait" href="trait.NotNull.html" title="diesel::sql_types::NotNull trait">NotNull</a></div><div class="item-right docblock-short"><p>A marker trait indicating that a SQL type is not null.</p>
</div><div class="item-left module-item"><a class="trait" href="trait.SingleValue.html" title="diesel::sql_types::SingleValue trait">SingleValue</a></div><div class="item-right docblock-short"><p>A marker trait indicating that a SQL type represents a single value, as
opposed to a list of values.</p>
</div><div class="item-left module-item"><a class="trait" href="trait.SqlOrd.html" title="diesel::sql_types::SqlOrd trait">SqlOrd</a></div><div class="item-right docblock-short"><p>Marker trait for types which can be used with <code>MAX</code> and <code>MIN</code></p>
</div><div class="item-left module-item"><a class="trait" href="trait.TypeMetadata.html" title="diesel::sql_types::TypeMetadata trait">TypeMetadata</a></div><div class="item-right docblock-short"><p>Information about how a backend stores metadata about given SQL types</p>
</div></div><h2 id="types" class="section-header"><a href="#types">Type Definitions</a></h2>
<div class="item-table"><div class="item-left module-item"><a class="type" href="type.Decimal.html" title="diesel::sql_types::Decimal type">Decimal</a></div><div class="item-right docblock-short"><p>Alias for <code>Numeric</code></p>
</div><div class="item-left module-item"><a class="type" href="type.VarChar.html" title="diesel::sql_types::VarChar type">VarChar</a></div><div class="item-right docblock-short"><p>The SQL <code>VARCHAR</code> type</p>
</div></div></section><section id="search" class="content hidden"></section><div id="rustdoc-vars" data-root-path="../../" data-current-crate="diesel" data-search-index-js="../../search-index.js" data-search-js="../../search.js"></div>
    <script src="../../main.js"></script>
</body></html>