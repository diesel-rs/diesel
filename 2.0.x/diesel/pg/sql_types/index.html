<!DOCTYPE html><html lang="en"><head><meta charset="utf-8"><meta name="viewport" content="width=device-width, initial-scale=1.0"><meta name="generator" content="rustdoc"><meta name="description" content="PostgreSQL specific SQL types"><title>diesel::pg::sql_types - Rust</title><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceSerif4-Regular-46f98efaafac5295.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/FiraSans-Regular-018c141bf0843ffd.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/FiraSans-Medium-8f9a781e4970d388.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceCodePro-Regular-562dcc5011b6de7d.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceSerif4-Bold-a2c9cd1067f8b328.ttf.woff2"><link rel="preload" as="font" type="font/woff2" crossorigin href="../../../static.files/SourceCodePro-Semibold-d899c5a5c4aeb14a.ttf.woff2"><link rel="stylesheet" href="../../../static.files/normalize-76eba96aa4d2e634.css"><link rel="stylesheet" href="../../../static.files/rustdoc-ba5701c5741a7b69.css" id="mainThemeStyle"><div id="rustdoc-vars" data-root-path="../../../" data-static-root-path="../../../static.files/" data-current-crate="diesel" data-themes="" data-resource-suffix="" data-rustdoc-version="1.71.0-nightly (7908a1d65 2023-04-17)" data-search-js="search-e077946657036a58.js" data-settings-js="settings-298e1ea74db45b39.js" data-settings-css="settings-7bfb4c59cc6bc502.css" data-theme-light-css="light-0f8c037637f9eb3e.css" data-theme-dark-css="dark-1097f8e92a01e3cf.css" data-theme-ayu-css="ayu-614652228113ac93.css" ></div><script src="../../../static.files/storage-62ce34ea385b278a.js"></script><script defer src="../../../static.files/main-1a524efa7bd4ab32.js"></script><noscript><link rel="stylesheet" media="(prefers-color-scheme:light)" href="../../../static.files/light-0f8c037637f9eb3e.css"><link rel="stylesheet" media="(prefers-color-scheme:dark)" href="../../../static.files/dark-1097f8e92a01e3cf.css"><link rel="stylesheet" href="../../../static.files/noscript-13285aec31fa243e.css"></noscript><link rel="alternate icon" type="image/png" href="../../../static.files/favicon-16x16-8b506e7a72182f1c.png"><link rel="alternate icon" type="image/png" href="../../../static.files/favicon-32x32-422f7d1d52889060.png"><link rel="icon" type="image/svg+xml" href="../../../static.files/favicon-2c020d218678b618.svg"></head><body class="rustdoc mod"><!--[if lte IE 11]><div class="warning">This old browser is unsupported and will most likely display funky things.</div><![endif]--><nav class="mobile-topbar"><button class="sidebar-menu-toggle">&#9776;</button><a class="logo-container" href="../../../diesel/index.html"><img class="rust-logo" src="../../../static.files/rust-logo-151179464ae7ed46.svg" alt="logo"></a><h2></h2></nav><nav class="sidebar"><a class="logo-container" href="../../../diesel/index.html"><img class="rust-logo" src="../../../static.files/rust-logo-151179464ae7ed46.svg" alt="logo"></a><h2 class="location"><a href="#">Module sql_types</a></h2><div class="sidebar-elems"><section><ul class="block"><li><a href="#structs">Structs</a></li><li><a href="#types">Type Definitions</a></li></ul></section></div></nav><main><div class="width-limiter"><nav class="sub"><form class="search-form"><span></span><input class="search-input" name="search" aria-label="Run search in the documentation" autocomplete="off" spellcheck="false" placeholder="Click or press ‘S’ to search, ‘?’ for more options…" type="search"><div id="help-button" title="help" tabindex="-1"><a href="../../../help.html">?</a></div><div id="settings-menu" tabindex="-1"><a href="../../../settings.html" title="settings"><img width="22" height="22" alt="Change settings" src="../../../static.files/wheel-7b819b6101059cd0.svg"></a></div></form></nav><section id="main-content" class="content"><div class="main-heading"><h1>Module <a href="../../index.html">diesel</a>::<wbr><a href="../index.html">pg</a>::<wbr><a class="mod" href="#">sql_types</a><button id="copy-path" title="Copy item path to clipboard"><img src="../../../static.files/clipboard-7571035ce49a181d.svg" width="19" height="18" alt="Copy item path"></button></h1><span class="out-of-band"><a class="srclink" href="../../../src/diesel/pg/types/mod.rs.html#28">source</a> · <button id="toggle-all-docs" title="collapse all docs">[<span>&#x2212;</span>]</button></span></div><span class="item-info"><div class="stab portability">Available on <strong>crate feature <code>postgres_backend</code></strong> only.</div></span><details class="toggle top-doc" open><summary class="hideme"><span>Expand description</span></summary><div class="docblock"><p>PostgreSQL specific SQL types</p>
<p>Note: All types in this module can be accessed through <code>diesel::sql_types</code></p>
</div></details><h2 id="structs" class="small-section-header"><a href="#structs">Structs</a></h2><ul class="item-table"><li><div class="item-name"><a class="struct" href="struct.Array.html" title="struct diesel::pg::sql_types::Array">Array</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/arrays.html"><code>Array</code></a> SQL type.</div></li><li><div class="item-name"><a class="struct" href="struct.Cidr.html" title="struct diesel::pg::sql_types::Cidr">Cidr</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/postgresql/static/datatype-net-types.html"><code>CIDR</code></a> SQL type. This type can only be used with <code>feature = &quot;network-address&quot;</code> or <code>feature = &quot;ipnet-address&quot;</code>.</div></li><li><div class="item-name"><a class="struct" href="struct.Inet.html" title="struct diesel::pg::sql_types::Inet">Inet</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/static/datatype-net-types.html"><code>INET</code></a> SQL type. This type can only be used with <code>feature = &quot;network-address&quot;</code> or <code>feature = &quot;ipnet-address&quot;</code>.</div></li><li><div class="item-name"><a class="struct" href="struct.Jsonb.html" title="struct diesel::pg::sql_types::Jsonb">Jsonb</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/datatype-json.html"><code>jsonb</code></a> SQL type.  This type can only be used with <code>feature = &quot;serde_json&quot;</code></div></li><li><div class="item-name"><a class="struct" href="struct.MacAddr.html" title="struct diesel::pg::sql_types::MacAddr">MacAddr</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/static/datatype-net-types.html"><code>MACADDR</code></a> SQL type.</div></li><li><div class="item-name"><a class="struct" href="struct.Money.html" title="struct diesel::pg::sql_types::Money">Money</a></div><div class="desc docblock-short">The PostgreSQL <a href="https://www.postgresql.org/docs/current/static/datatype-money.html">Money</a> type.</div></li><li><div class="item-name"><a class="struct" href="struct.Oid.html" title="struct diesel::pg::sql_types::Oid">Oid</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/datatype-oid.html"><code>OID</code></a> SQL type. This is a PostgreSQL specific type.</div></li><li><div class="item-name"><a class="struct" href="struct.Range.html" title="struct diesel::pg::sql_types::Range">Range</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/rangetypes.html"><code>Range</code></a> SQL type.</div></li><li><div class="item-name"><a class="struct" href="struct.Record.html" title="struct diesel::pg::sql_types::Record">Record</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/rowtypes.html"><code>Record</code></a> (a.k.a. tuple) SQL type.</div></li><li><div class="item-name"><a class="struct" href="struct.Timestamptz.html" title="struct diesel::pg::sql_types::Timestamptz">Timestamptz</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/datatype-datetime.html">“timestamp with time zone” SQL type</a>, which PostgreSQL abbreviates
to <code>timestamptz</code>.</div></li><li><div class="item-name"><a class="struct" href="struct.Uuid.html" title="struct diesel::pg::sql_types::Uuid">Uuid</a></div><div class="desc docblock-short">The <a href="https://www.postgresql.org/docs/current/datatype-uuid.html"><code>UUID</code></a> SQL type. This type can only be used with <code>feature = &quot;uuid&quot;</code></div></li></ul><h2 id="types" class="small-section-header"><a href="#types">Type Definitions</a></h2><ul class="item-table"><li><div class="item-name"><a class="type" href="type.BigSerial.html" title="type diesel::pg::sql_types::BigSerial">BigSerial</a></div><div class="desc docblock-short">Alias for <a href="../../sql_types/struct.BigInt.html" title="struct diesel::sql_types::BigInt"><code>BigInt</code></a></div></li><li><div class="item-name"><a class="type" href="type.Serial.html" title="type diesel::pg::sql_types::Serial">Serial</a></div><div class="desc docblock-short">Alias for <a href="../../sql_types/struct.Integer.html" title="struct diesel::sql_types::Integer"><code>Integer</code></a></div></li><li><div class="item-name"><a class="type" href="type.SmallSerial.html" title="type diesel::pg::sql_types::SmallSerial">SmallSerial</a></div><div class="desc docblock-short">Alias for <a href="../../sql_types/struct.SmallInt.html" title="struct diesel::sql_types::SmallInt"><code>SmallInt</code></a></div></li></ul></section></div></main></body></html>