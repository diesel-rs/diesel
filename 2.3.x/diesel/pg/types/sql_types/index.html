<!DOCTYPE html>
<html lang="en">
<head>
    <meta http-equiv="refresh" content="0;URL=../../../../diesel/pg/sql_types/index.html">
    <title>Redirection</title>
</head>
<body>
    <p>Redirecting to <a href="../../../../diesel/pg/sql_types/index.html">../../../../diesel/pg/sql_types/index.html</a>...</p>
    <script>location.replace("../../../../diesel/pg/sql_types/index.html" + location.search + location.hash);</script>
</body>
</html>