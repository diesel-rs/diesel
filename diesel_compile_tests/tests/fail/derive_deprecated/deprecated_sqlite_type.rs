extern crate diesel;

use diesel::sql_types::SqlType;

#[derive(SqlType)]
#[sqlite_type]
//~^ ERROR: unexpected end of input, expected `=`
struct Type1;

#[derive(SqlType)]
#[sqlite_type()]
//~^ ERROR: expected `=`
struct Type2;

#[derive(SqlType)]
#[sqlite_type = 1]
//~^ ERROR: expected string literal
struct Type3;

#[derive(SqlType)]
//~^ ERROR: no variant or associated item named `foo` found for enum `SqliteType` in the current scope
#[sqlite_type = "foo"]
struct Type4;

fn main() {}
