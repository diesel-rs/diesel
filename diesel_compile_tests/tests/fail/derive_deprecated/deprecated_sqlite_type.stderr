error: unexpected end of input, expected `=`
       help: The correct format looks like `#[diesel(sqlite_type(name = "foo"))]`
 --> tests/fail/derive_deprecated/deprecated_sqlite_type.rs:6:3
  |
6 | #[sqlite_type]
  |   ^^^^^^^^^^^

error: expected `=`
  --> tests/fail/derive_deprecated/deprecated_sqlite_type.rs:10:14
   |
10 | #[sqlite_type()]
   |              ^^

error: expected string literal
  --> tests/fail/derive_deprecated/deprecated_sqlite_type.rs:14:17
   |
14 | #[sqlite_type = 1]
   |                 ^

error[E0599]: no variant or associated item named `foo` found for enum `SqliteType` in the current scope
  --> tests/fail/derive_deprecated/deprecated_sqlite_type.rs:17:10
   |
17 | #[derive(SqlType)]
   |          ^^^^^^^ variant or associated item not found in `SqliteType`
   |
   = note: this error originates in the derive macro `SqlType` (in Nightly builds, run with -Z macro-backtrace for more info)
